src/main/java/com/google/cloud/tools/maven/Deploy.java

```diff
@@ -0,0 +1,186 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.tools.maven;
+
+import com.google.cloud.tools.Action;
+import com.google.cloud.tools.StageGenericJavaAction;
+import com.google.cloud.tools.maven.configs.DeployConfig;
+import com.google.cloud.tools.maven.configs.StageConfig;
+import com.google.cloud.tools.StageAction;
+import com.google.common.base.Strings;
+import com.google.cloud.tools.DeployAction;
+import com.google.cloud.tools.DeployAction.AppType;
+import com.google.cloud.tools.InvalidDirectoryException;
+import com.google.cloud.tools.InvalidFlagException;
+import com.google.cloud.tools.Option;
+
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.plugins.annotations.Component;
+import org.apache.maven.plugins.annotations.Execute;
+import org.apache.maven.plugins.annotations.LifecyclePhase;
+import org.apache.maven.plugins.annotations.Mojo;
+import org.apache.maven.plugins.annotations.Parameter;
+import org.codehaus.plexus.util.FileUtils;
+import org.eclipse.aether.RepositorySystem;
+import org.eclipse.aether.RepositorySystemSession;
+import org.eclipse.aether.repository.RemoteRepository;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Deploys an app to the server.
+ */
+@Mojo(name = "deploy")
+@Execute(phase = LifecyclePhase.PACKAGE)
+public class Deploy extends GcpAppMojo {
+
+  // Maven parameters.
+  @Parameter(defaultValue = "${stage}")
+  private StageConfig stageConfig;
+  @Parameter(defaultValue = "${deploy}")
+  private DeployConfig deployConfig;
+  @Parameter(property = "gcp.app.deploy.stagingDir",
+      defaultValue = "${project.build.directory}/appengine-staging")
+  private String stagingDirLocation;
+  @Parameter(property = "gcp.app.deploy.sourceDir",
+      defaultValue = "${project.build.directory}/${project.build.finalName}")
+  private String sourceDirLocation;
+  @Parameter(property = "gcp.app.deploy.configDir",
+      defaultValue = "${project.build.directory}/src/main/appengine")
+  private String configLocation;
+
+  // Required by SdkResolver.
+  @Component
+  protected RepositorySystem repositorySystem;
+  @Parameter(defaultValue = "${repositorySystemSession}")
+  protected RepositorySystemSession repositorySession;
+  @Parameter(defaultValue = "${project.remoteProjectRepositories}", readonly = true)
+  protected List<RemoteRepository> projectRepositories;
+  @Parameter(defaultValue = "${project.remotePluginRepositories}", readonly = true)
+  protected List<RemoteRepository> pluginRepositories;
+
+  // Class members.
+  private File stagingDir;
+
+  public void execute() throws MojoExecutionException {
+    getLog().info(String.format("Starting to execute app to project %s.", cloudProject));
```

Comment:
I thought it was a best practice and more readable.

Is "getLog().info("Starting to execute app to project " + cloudProject + ".");" better for you?